tests/ui/const-generics/occurs-check/unused-substs-1.stderr

```diff
@@ -4,7 +4,7 @@ error[E0277]: the trait bound `A<_>: Bar<_>` is not satisfied
 LL |     let _ = A;
    |             ^ the trait `Bar<_>` is not implemented for `A<_>`
    |
-   = help: the trait `Bar<N>` is implemented for `A<7>`
+   = help: the trait `Bar<_>` is implemented for `A<7>`
```

Comment:
While this case isn't that much worse, there are cases where it shows something like `UwU<_> is implemented for OwO<_>` which no longer indicates that the type that's being printed as `_` is the same in both cases. (I dislike that that sort of printing for types is *ever* done, but that's not specific to this PR.)

Reply:
Yeah, I wonder if we could plug up the remaining infer vars here, so that those two `Owo<_>` and `Uwu<_>` are re-connected with some placeholder name.